django/contrib/admin/sites.py

```diff
@@ -90,7 +90,15 @@ def register(self, model_or_iterable, admin_class=None, **options):
                 )
 
             if model in self._registry:
-                raise AlreadyRegistered('The model %s is already registered' % model.__name__)
+                registered_admin = self._registry[model]
+                registered_model = registered_admin.model
+                raise AlreadyRegistered(
+                    'Cannot register %s for %s, because %s is already registered.' % (
```

Comment:
The wording doesn't seem quite right: "<model> for <registered admin>" seems like it should be "Cannot register <model> because <model> is already registered with <registered admin>."